### PR TITLE
Added event logging in database

### PR DESCRIPTION
**Pull request by fritsjustbetter**

### CODE REVIEW

File: database/migrations/2024_01_24_153000_create_event_log_table.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('event_logs', function (Blueprint $table): void {
```
Comment (VincentBean) [inline database/migrations/2024_01_24_153000_create_event_log_table.php]: Please prefix `magento_webhooks`

**Review by VincentBean — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -34,13 +34,30 @@ Publish the configuration of the package.
...
 
+### Schedule clean up command
+
+In order to keep the database from filling up, schedule the `CleanLogsCommand` in your Kernel.
+
+```php
+$schedule->command(CleanLogsCommand::class)->daily();
+```
+
+This command supports usage of a custom date, this date will indicate removing the logs from before that date, this defaults to one month.
```
Comment (ramonrietdijk) [inline README.md]: Please, rewrite this sentence. Also, the `--force` option does not exist.

File: UPGRADE.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Upgrade guide
+
+# 1.x to 2.x
```
Comment (ramonrietdijk) [inline UPGRADE.md]: The type of headings are incorrect. Use `#`, `##` and `###`

File: src/Models/EventLog.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+namespace JustBetter\MagentoWebhooks\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Carbon;
+
+/**
+ * @property int $id
+ * @property string $event
+ * @property ?array $data
+ * @property Carbon $created_at
```
Comment (ramonrietdijk) [inline src/Models/EventLog.php]: Both `created_at` and `updated_at` are nullable

File: tests/Actions/CleanLogsTest.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Actions;
```
Comment (ramonrietdijk) [inline tests/Actions/CleanLogsTest.php]: Namespace is incorrect

File: tests/Commands/CommandDispatchTest.php
Comment (ramonrietdijk) [inline tests/Commands/CommandDispatchTest.php:1]: Not a great name for a test, please rename to `CleanLogsCommandTest`

File: tests/Commands/CommandDispatchTest.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+use Illuminate\Support\Facades\Bus;
+use Illuminate\Testing\PendingCommand;
+use JustBetter\MagentoWebhooks\Commands\CleanLogsCommand;
+use JustBetter\MagentoWebhooks\Jobs\CleanLogsJob;
+use JustBetter\MagentoWebhooks\Tests\TestCase;
+
+class CommandDispatchTest extends TestCase
+{
+    /** @test */
+    public function test_it_dispatches_job(): void
```
Comment (ramonrietdijk) [inline tests/Commands/CommandDispatchTest.php]: It's helpful to have more descriptive test cases as it currently doesn't tell that much. Also, the `test_` prefix is not required with the `@test` comment.

File: tests/Jobs/CleanLogsJobTest.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Jobs;
```
Comment (ramonrietdijk) [inline tests/Jobs/CleanLogsJobTest.php]: Namespace is incorrect

File: tests/Jobs/CleanLogsJobTest.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+use JustBetter\MagentoWebhooks\Contracts\CleansLogs;
+use JustBetter\MagentoWebhooks\Jobs\CleanLogsJob;
+use JustBetter\MagentoWebhooks\Tests\TestCase;
+use Mockery\MockInterface;
+
+class CleanLogsJobTest extends TestCase
+{
+    /** @test */
+    public function it_can_dispatch_events(): void
```
Comment (ramonrietdijk) [inline tests/Jobs/CleanLogsJobTest.php]: Test case does not seem correct

**Review by ramonrietdijk — CHANGES_REQUESTED** [review]

**Review by ramonrietdijk — APPROVED** [review]

**Comment by VincentBean** [comment]
Please fix the failing tests :)

**Review by VincentBean — APPROVED** [review]

**Pull request merged**